core/blockchain.go

```diff
@@ -313,6 +321,12 @@ func NewBlockChain(db ethdb.Database, cacheConfig *CacheConfig, genesis *Genesis
 
 	// Declare the end of the verkle transition if need be
 	if bc.chainConfig.Rules(head.Number, false /* XXX */, head.Time).IsPrague {
+		// TODO this only works when resuming a chain that has already gone
+		// through the conversion. All pointers should be saved to the DB
+		// for it to be able to recover if interrupted during the transition
+		// but that's left out to a later PR since there's not really a need
+		// right now.
+		bc.stateCache.InitTransitionStatus(true, true)
```

Comment:
I think this has caused some confusions and problems before, but we'll see.

Reply:
it works for all tests that I'm aware of (shadowfork, post-genesis transition and kaustinen sync), but I agree it's tacky. There is a TODO however, so this will be fixed eventually.